packages/lmi/src/lmi/llms.py

```diff
@@ -527,13 +533,20 @@ class LiteLLMModel(LLMModel):
 
     @model_validator(mode="before")
     @classmethod
-    def maybe_set_config_attribute(cls, data: dict[str, Any]) -> dict[str, Any]:
+    def maybe_set_config_attribute(cls, input_data: dict[str, Any]) -> dict[str, Any]:
```

Comment:
We have hit this issue before where our validators mutate the inputs, leaving side effects. We should type the `input_data` with `Mapping` to prevent this issue